lib/handsoap/service.rb

```diff
@@ -96,9 +96,10 @@ class Service
     @@create_document_callback = nil
     @@log_header_callback = nil
     @@log_body_callback = nil
-    attr_reader :http_client
+    attr_reader :http_client, :cookie
     def initialize(ep)
       @http_client = nil
+      @cookie = nil
```

Comment:
That would work. You wouldn't even need to pass in the headers as a separate arg, just pass in the entire http response, then the class could extract what ever it needs.